### ISSUE TITLE
 Network - Resource request error: QNetworkReply::NetworkError(HostNotFoundError) ( "Host awesome.dev not found" )

### ISSUE THREAD

**Issue by svenbluege** [ariya/phantomjs#14272]
**Which version of PhantomJS are you using? Tip: run `phantomjs --version`.**

2.1.1

**What steps will reproduce the problem?**
1. start phantomjs with `--webdriver=4444 --debug=true`
2. get a session id `curl -H "Accept: application/json" -H "Content-Type: application/json;charset=UTF-8" --data "{\"desiredCapabilities\":{\"unexpectedAlertBehaviour\":\"accept\",\"webStorageEnabled\":true,\"browserName\":\"phantomjs\",\"proxy\":{\"httpProxy\":\"none\",\"proxyType\":\"manual\"}}}" http://localhost:4444/wd/hub/session`
3. request a remote url `curl --data "{\"url\":\"http:\/\/10.0.121.82\/tests\/joomla-cms3\/installation\/index.php\"}" http://localhost:4444/wd/hub/session/2a11c810-175b-11e6-a3e8-03a5c58922d4/url`
4. see the error message in the phantomjs windows: 

```
2016-05-11T11:32:11 [DEBUG] HTTP Request - HTTP Version 1.1
2016-05-11T11:32:11 [DEBUG] HTTP Request - Query String
2016-05-11T11:32:11 [DEBUG] HTTP Request - Receiving Header "User-Agent" = "curl/7.33.0"
2016-05-11T11:32:11 [DEBUG] HTTP Request - Receiving Header "Host" = "localhost:4444"
2016-05-11T11:32:11 [DEBUG] HTTP Request - Receiving Header "Accept" = "*/*"
2016-05-11T11:32:11 [DEBUG] HTTP Request - Receiving Header "Content-Length" = "75"
2016-05-11T11:32:11 [DEBUG] HTTP Request - Receiving Header "Content-Type" = "application/x-www-form-urlencoded"
2016-05-11T11:32:11 [DEBUG] HTTP Request - Method POST/PUT
2016-05-11T11:32:11 [DEBUG] HTTP Request - Content Body: {"url":"http:\/\/10.0.121.82\/tests\/joomla-cms3\/installation\/index.php"}
2016-05-11T11:32:11 [DEBUG] WebPage - updateLoadingProgress: 10
2016-05-11T11:32:13 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(HostNotFoundError) ( "Host 10.0.121.82 not found" ) URL: "http://10.0.121.82/tests/joomla-cms3/installation/index.php"
2016-05-11T11:32:13 [DEBUG] WebPage - updateLoadingProgress: 100
2016-05-11T11:32:13 [DEBUG] WebPage - setupFrame ""
2016-05-11T11:32:13 [DEBUG] WebPage - setupFrame ""
2016-05-11T11:32:13 [DEBUG] HTTP Response - Status Code 200 OK
2016-05-11T11:32:13 [DEBUG] HTTP Response - Sending Header "Cache" = "no-cache"
2016-05-11T11:32:13 [DEBUG] HTTP Response - Sending Header "Content-Length" = "74"
2016-05-11T11:32:13 [DEBUG] HTTP Response - Sending Header "Content-Type" = "application/json;charset=UTF-8"
```

the most important message is this: `Network - Resource request error: QNetworkReply::NetworkError(HostNotFoundError) ( "Host 10.0.121.82 not found" )`. The connection works perfectly with localhost or 127.0.0.1. But if I add another IP or host name the request fails with `HostNotFoundError`.

**Which operating system are you using?**

Windows 7

**Did you use binary PhantomJS or did you compile it from source?**

Binary

**Please provide any additional information below.**
the steps to repeat do use curl only. Running a request to the same url from a javascript file called like `phantomjs test.js` works fine.

**Comment by Vitallium** [comment]
Perhaps something is wrong with your network?

**Comment by svenbluege** [comment]
That's possible. But in this case I would expect the same behavior for both the normal phantomjs call and the webdriver mode. I just does not work in webdriver mode. Is there something different I overlooked?

**Comment by dmsmidt** [comment]
I had the same error, and only for locally hosted sites. While the URL worked in a normal browser PhantomJS couldn't work it out. 
I retried via a wildcard DNS service (http://xip.io), and suddenly it worked. So it seems some networking/permission thing. 
Any idea's what can be the problem?

**Comment by dmsmidt** [comment]
I managed to solve my problem. The difference is that I used a domain  name as a host and the original question is about an IP address. Still I think it a network issue.

A good way to check if a domain name can be resolved is using the following command (unix):
`getent hosts some.domain`. If it doesn't return anything, but `dig some.domain` does, you probably have to change the `hosts` config in `/etc/nsswitch.conf`. I had to make `resolve` the first item, since my resolve file defines the DNS server to use.
